src/index.tsx

```diff
@@ -1,251 +1,276 @@
-					const character = value.charAt(i)
-					if (character == character.toLowerCase()) {
-						// Character is lowercase, numeric, or a symbol
-					} else if (character == character.toUpperCase()) {
-						return true
-					}
-					i++
-				}
-				return false
-			})(),
-			message: messages.capital || "Password has a capital letter.",
-		},
-		match: {
-			valid: value.length > 0 && value === valueAgain,
-			message: messages.match || "Passwords match.",
-		},
-		lowercase: {
-			valid: (() => {
-				var i = 0
-				if (value.length === 0) {
-					return false
-				}
-				while (i < value.length) {
-					const character = value.charAt(i)
-					if (character == character.toUpperCase()) {
-						// Character is lowercase, numeric, or a symbol
-					} else if (character == character.toLowerCase()) {
-						return true
-					}
-					i++
-				}
-				return false
-			})(),
-			message: messages.lowercase || "Password has a lowercase letter.",
-		},
-		letter: {
-			valid: /[a-zA-Z]/g.test(value),
-			message: messages.letter || "Password has a letter.",
-		},
-		maxLength: {
-			valid: value.length <= (maxLength || 16),
-			message: messages.maxLength || `Password has no more than ${maxLength} characters.`,
-		},
-		notEmpty: {
-			valid: Boolean(value.length > 0 && valueAgain && valueAgain.length > 0),
-			message: messages.notEmpty || "Password fields are not empty.",
-		},
-	}
-	const enabledRules: RuleNames[] = useMemo(
-		() => rules.filter((rule) => Boolean(ruleDefinitions[rule])),
-		[rules],
-	)
-	const isValid: boolean = enabledRules.every((rule) => ruleDefinitions[rule].valid)
-	const failedRules: RuleNames[] = useMemo(
-		() => enabledRules.filter((rule) => !ruleDefinitions[rule].valid),
-		[value, valueAgain, enabledRules],
-	)
+  const ruleDefinitions: {
+    [key in RuleNames]: { valid: boolean; message: string }
+  } = {
+    minLength: {
+      valid: value.length >= (minLength || 100),
+      message:
+        messages.minLength || `Password has at least ${minLength} characters.`
+    },
+    specialChar: {
+      valid: specialCharsRegex.test(value),
+      message: messages.specialChar || "Password has special characters."
+    },
+    number: {
+      valid: /\d/g.test(value),
+      message: messages.number || "Password has a number."
+    },
+    match: {
+      valid: value.length > 0 && value === valueAgain,
+      message: messages.match || "Passwords match."
+    },
+    capital: {
+      valid: valueHasCapitalCharacters(value),
+      message: messages.capital || "Password has a capital letter."
+    },
+    lowercase: {
+      valid: valueHasLowercaseCharacters(value),
+      message: messages.lowercase || "Password has a lowercase letter."
+    },
+    capitalAndLowerCase: {
+      valid:
+        valueHasCapitalCharacters(value) && valueHasLowercaseCharacters(value),
+      message:
+        messages.capitalAndLowerCase ||
+        "Password Contains at least one uppercase and lowercase characters"
```

Comment:
Please use lowercase "c" in `contains` (both on this line and in the `noSpaces` rule)

Reply:
@sators Done ✅